### PR TITLE
Spring 03

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

**Review by Likh-Alex — APPROVED** [review]

File: src/main/java/spring/practice/dao/UserDao.java [hunk 1]
```diff
@@ -7,4 +7,6 @@
     void add(User user);
 
     List<User> listUsers();
+
+    User get(Long id);
```
Comment (Sofasmile) [inline src/main/java/spring/practice/dao/UserDao.java]:
```suggestion
    Optional<User> get(Long id);
```

File: src/main/java/spring/practice/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -49,4 +49,16 @@ public void add(User user) {
             throw new RuntimeException("Can't get list of Users", e);
         }
     }
+
+    @Override
+    public User get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            Query<User> getUserByIdQuery = session.createQuery("FROM User u "
```
Comment (Sofasmile) [inline src/main/java/spring/practice/dao/impl/UserDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/04_spring/checklist/hw_03.html

> Don’t complicate get by id methods in your DAO, use session.get().

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/spring/practice/dao/UserDao.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/spring/practice/dao/UserDao.java]: refactored

File: src/main/java/spring/practice/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/spring/practice/dao/impl/UserDaoImpl.java]: refactored due to changing `get` method

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**